Source/TextToSpeechV1/TextToSpeech.swift

```diff
@@ -309,38 +291,39 @@ public class TextToSpeech {
         )
 
         // execute REST request
-        request.responseData {
-            (response: RestResponse<Data>) in
-            switch response.result {
-            case .success(let data):
-                if accept?.lowercased().contains("audio/wav") == true {
-                    // repair the WAV header
-                    var wav = data
-                    guard WAVRepair.isWAVFile(data: wav) else {
-                        let failureReason = "Returned audio is in an unexpected format."
-                        let userInfo = [NSLocalizedDescriptionKey: failureReason]
-                        let error = NSError(domain: self.domain, code: 0, userInfo: userInfo)
-                        failure?(error)
-                        return
-                    }
-                    WAVRepair.repairWAVHeader(data: &wav)
-                    success(wav)
-                } else if accept?.lowercased().contains("ogg") == true && accept?.lowercased().contains("opus") == true {
-                    do {
-                        let decodedAudio = try TextToSpeechDecoder(audioData: data)
-                        success(decodedAudio.pcmDataWithHeaders)
-                    } catch {
-                        let failureReason = "Returned audio is in an unexpected format."
-                        let userInfo = [NSLocalizedDescriptionKey: failureReason]
-                        let error = NSError(domain: self.domain, code: 0, userInfo: userInfo)
-                        failure?(error)
-                        return
-                    }
-                } else {
-                    success(data)
+        request.responseObject { (response: WatsonResponse<Data>?, error: Error?) in
+            var newResponse = response
+            guard let data = response?.result else {
+                completionHandler(nil, error)
+                return
+            }
+            if accept?.lowercased().contains("audio/wav") == true {
+                // repair the WAV header
+                var wav = data
+                guard WAVRepair.isWAVFile(data: wav) else {
+                    let failureReason = "Returned audio is in an unexpected format."
+                    let userInfo = [NSLocalizedDescriptionKey: failureReason]
+                    let error = NSError(domain: self.domain, code: 0, userInfo: userInfo)
+                    completionHandler(nil, error)
```

Comment:
Since these lines are changing anyway, let's replace the `NSError` with the more Swifty `Error`, e.g.
```
let error = Error("Returned audio is in an unexpected format.")
```